website/server/models/user/methods.js

```diff
@@ -20,6 +20,49 @@ schema.methods.getGroups = function getUserGroups () {
   return userGroups;
 };
 
+// Get an array of error message keys that would be thrown if the given interaction was attempted
```

Comment:
Technically, if the given interaction was attempted, only one error message would be thrown.  Is it really necessary to return ALL issues with an interaction?  Why not just the first one found (if a problem exists) or none (if the interaction is allowed)?